src/gf.c

```diff
@@ -601,7 +601,7 @@ static void jl_compilation_sig(
     for (i = 0; i < np; i++) {
         jl_value_t *elt = jl_tparam(tt, i);
         jl_value_t *decl_i = jl_nth_slot_type(decl, i);
-        size_t i_arg = (i <= nargs ? i : nargs);
+        size_t i_arg = (i < nargs - 1 ? i : nargs - 1);
```

Comment:
`min(i, nargs - 1)` for readability?

Reply:
Readability+C? You're funny. The `min` function can't exist in C since it doesn't have templates, and `fmin` is a floating point function.